datadog_checks_base/datadog_checks/base/checks/win/winpdh_base.py

```diff
@@ -182,3 +138,51 @@ def check(self, instance):
             except Exception as e:
                 # don't give up on all of the metrics because one failed
                 self.log.error("Failed to get data for %s %s: %s" % (inst_name, dd_name, str(e)))
+
+    def _make_counters(self, key, counter_data):
+        counter_list, (datatypes, remote_machine, check_instance, message) = counter_data
+
+        # list of the metrics. Each entry is itself an entry,
+        # which is the pdh name, datadog metric name, type, and the
+        # pdh counter object
+        for counterset, inst_name, counter_name, dd_name, mtype in counter_list:
+            if check_instance and self._no_instance(inst_name):
+                inst_name = None
+
+            m = getattr(self, mtype.lower())
+            precision = datatypes.get(dd_name)
+
+            try:
+                obj = WinPDHCounter(
+                    counterset,
+                    counter_name,
+                    self.log,
+                    inst_name,
+                    machine_name=remote_machine,
+                    precision=precision
+                )
+            except Exception as e:
+                self.log.debug(
+                    'Could not create counter {}\\{} due to {}, will not report {}.'.format(
+                        counterset, counter_name, e, dd_name
+                    )
+                )
+                self._missing_counters[(counterset, inst_name, counter_name, dd_name, mtype)] = (
+                    datatypes, remote_machine, check_instance, message
+                )
+                continue
+            else:
+                self._missing_counters.pop((counterset, inst_name, counter_name, dd_name, mtype), None)
+
+            entry = [inst_name, dd_name, m, obj]
+            self.log.debug('{}: {}'.format(message, entry))
+            self._metrics[key].append(entry)
```

Comment:
Because of this method usage at https://github.com/DataDog/integrations-core/pull/2654/files#diff-14769541de478cc114862b41f3cdcfe2R122 , isn't it a risk that the key won't exist?

Reply:
Key will always exist https://github.com/DataDog/integrations-core/blob/8567b6eae5ad8c5c16a96b5d35f379ab6a2cbcb6/datadog_checks_base/datadog_checks/base/checks/win/winpdh_base.py#L55